src/types/message-types.ts

```diff
@@ -75,6 +76,13 @@ export type QueryResultEntry = GenericMessage & {
   encodedData?: string;
 };
 
+export type MessageSubscriptionHandler = (message: GenericMessage) => void;
+
+export interface GenericMessageSubscription {
```

Comment:
Paired with the earlier comment, consider:
```suggestion
export interface MessageSubscription {
```